ci-operator/step-registry/baremetalds/devscripts/gather/baremetalds-devscripts-gather-commands.sh

```diff
@@ -55,20 +31,65 @@ echo "Get install-gather, if there is one..."
 cp /root/dev-scripts/ocp/*/log-bundle*.tar.gz /tmp/artifacts/log-bundle-\$HOSTNAME.tar.gz || true
 
 echo "Get sosreport including sar data..."
-sos report --case-id "\$HOSTNAME" --batch -o container_log,filesys,kvm,libvirt,logs,networkmanager,podman,processor,rpm,sar,virsh,yum --tmp-dir /tmp/artifacts
+sos report --case-id "\$HOSTNAME" --batch \
+  -o container_log,filesys,kvm,libvirt,logs,networkmanager,podman,processor,rpm,sar,virsh \
+  -k podman.all -k podman.logs \
+  --tmp-dir /tmp/artifacts
 
 echo "Get libvirt logs..."
 tar -czC "/var/log/libvirt/qemu" -f "/tmp/artifacts/libvirt-logs-\$HOSTNAME.tar.gz" --transform "s?^\.?libvirt-logs-\$HOSTNAME?" .
 
+. common.sh
+. network.sh
+. utils.sh
+
+# Pass master IPs to installer-gather script to collect info from nodes which didn't join the cluster
+NODE_IPS=()
+for (( n=0; n<\$NUM_MASTERS; n++ ))
+do
+    node_name=\$(printf \$MASTER_HOSTNAME_FORMAT \$n)
+    node_ip=\$(sudo virsh net-dumpxml \$BAREMETAL_NETWORK_NAME | xmllint --xpath "string(//ip/dhcp[*]/host[@name='\${node_name}']/@ip)" -)
```

Comment:
Q: It's not clear to me how to determine which are the nodes that didn't join the cluster, @vrutkovs can you please clarify? Looks like sos report is exercised against all the nodes found, regardless their state? Thanks

(IIRC, the `installer-gather.sh` script is present only if there's a failure and so in that case the node effectively didn't join the cluster; the `|| true` should cover the positive case instead)

Reply:
ah, xmllint will return an empty IP? That can be updated in a followup PR to make this cleaner imo

Reply:
> Looks like sos report is exercised against all the nodes found, regardless their state? 

Yes, exactly - we want to see why some nodes are known to libvirt but haven't joined the cluster

Reply:
Isn't that generating unnecessary artifacts for successfull builds? It would be nice to have it constrained only for the "bad" nodes, even though I'm not sure about the criteria to be adopted to uniquely identify them (a simple heuristic maybe could be the presence of the `installer-gather.sh` script?)

Reply:
> ah, xmllint will return an empty IP? That can be updated in a followup PR to make this cleaner imo

If not wrong, unfortunately. There was a leftover for static IP builds where the previously DHCP addresses where not removed, and this could be misleading (for humans too I'd say) (cc @bfournie )

I've just double-checked the [agent-compact-ipv4 build](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/34664/rehearse-34664-pull-ci-openshift-installer-master-e2e-agent-compact-ipv4/1663458871177383936/artifacts/e2e-agent-compact-ipv4/baremetalds-devscripts-gather/build-log.txt), which uses static IPs, and it's currently failing for that reason (if you look at the related sos reports, you could see that the correct ips are 192.168.111.80, .81, and .82 respectively). As mentioned before, in general the `<dns>` tag looks more reliable IMHO

Reply:
@vrutkovs has this point already been addressed?

Reply:
Good catch, updated